src/libraries/System.Reflection.MetadataLoadContext/src/System/Reflection/TypeLoading/General/CoreTypes.cs

```diff
@@ -43,7 +43,7 @@ internal CoreTypes(MetadataLoadContext loader, string coreAssemblyName)
         /// <summary>
         /// Returns null if the specific core type did not exist or could not be loaded. Call GetException(coreType) to get detailed info.
         /// </summary>
-        public RoType this[CoreType coreType] => _coreTypes[(int)coreType];
-        public Exception GetException(CoreType coreType) => _exceptions[(int)coreType];
+        public RoType? this[CoreType coreType] => _coreTypes[(int)coreType];
+        public Exception? GetException(CoreType coreType) => _exceptions[(int)coreType];
```

Comment:
NIT: Seems _exceptions not need to have nullable elements, then this would return non null and a caller doesn't need to bang it

Reply:
Are you sure? It seems to me that it's more likely for it to contain values than not.

Reply:
I meant a null Exception would never added to `_exceptions[]`, in the constructor if `type` is null `e` wouldn't be null, but yes compiler would still warn for `e` and we should bang that, just for correctness. Anyways it is not affecting public API and not making much difference so I commented as NIT, not have to change 
```csharp
for (int i = 0; i < numCoreTypes; i++) 
{ 
    ((CoreType)i).GetFullName(out byte[] ns, out byte[] name); 
    RoType type = coreAssembly.GetTypeCore(ns, name, ignoreCase: false, out e); 
    coreTypes[i] = type; 
    if (type == null) 
    { 
        exceptions[i] = e;
    }
}
```